helper/schema/schema_test.go

```diff
@@ -5991,7 +5991,33 @@ func TestSchemaMap_Validate(t *testing.T) {
 
 			Err: true,
 			Errors: []error{
-				fmt.Errorf(`Error: ConflictsWith: "blacklist": conflicts with whitelist`),
+				fmt.Errorf(`Error: ConflictsWith:  "blacklist": conflicts with whitelist`),
+			},
+		},
+
+		"Conflicting attributes generate warning": {
+			Schema: map[string]*Schema{
+				"whitelist": {
+					Type:     TypeString,
+					Optional: true,
+				},
+				"blacklist": {
+					Type:              TypeString,
+					Optional:          true,
+					ConflictsWith:     []string{"whitelist"},
+					ConditionsMode:    "warning",
+					ConditionsMessage: "This functionality will be removed in a later release.",
+				},
+			},
+
+			Config: map[string]interface{}{
+				"whitelist": "white-val",
+				"blacklist": "black-val",
```

Comment:
nitpick: can we use "allowlist" and "denylist" or something similar?

Reply:
(I'm sure you're just copying code that exists, but I'd like to leave the campsite better than we found it and not carry that pattern into the future. :) )

Reply:
I would prefer this too! Would it be ok if I replace this in other tests too? Or should I limit the scope of my changes for this PR? (For reference, there are 144 instances of "whitelist" in this file).

Reply:
As long as the tests pass, I would be happy to accept that change as part of this PR, in the spirit of leaving the area of code we're working on better than we found it, as we're unlikely to prioritize a PR that changed only those things. :)